test/notifications/display_test.dart

```diff
@@ -131,6 +131,59 @@ void main() {
             NotificationChannelManager.kVibrationPattern)
       ;
     });
+
+    test('channel is not recreated if one with same id already exists', () async {
+      await NotificationChannelManager.ensureChannel();
+      check(testBinding.androidNotificationHost.takeChannelMethodCallLogs())
+        .deepEquals(['getNotificationChannels', 'createNotificationChannel']);
+
+      await NotificationChannelManager.ensureChannel();
```

Comment:
It looks like the first `ensureChannel` here is to simulate a channel having gotten created in the past, and then the second is the call that we're testing the behavior of.

I think it'd be cleaner to do the setup more directly, without `ensureChannel`. Using `ensureChannel` will only produce a channel that's exactly the way the current code sets it up; this breaks down in the next test case, where we want to simulate some previous version of the app having set it up, and it also breaks down with the possibility that the user may have altered the settings on the channel. (If they have, we should leave the existing channel in place; only the ID counts for saying it's the same channel.)